### PR TITLE
parser: Handle MySQL-specific code

### PR DESCRIPTION
**Pull request by shenli**
Fix: https://github.com/pingcap/tidb/issues/971

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: parser/yy_parser.go [hunk 1]
```diff
@@ -30,6 +33,31 @@ const (
 	CodeSyntaxErr terror.ErrCode = 1
 )
 
+var (
+	specCodePattern = regexp.MustCompile(`\/\*!(M?[0-9]{5,6})?([^*]|\*+[^*/])*\*+\/`)
+	specCodeStart   = regexp.MustCompile(`^\/\*!(M?[0-9]{5,6} )?`)
```
Comment (zimulala) [inline parser/yy_parser.go]: `^\/\*!(M?[0-9]{5,6} )?` -> `^\/\*!(M?[0-9]{5,6}[ \t])?` ？

File: parser/yy_parser.go [hunk 2]
```diff
@@ -30,6 +33,31 @@ const (
...
+// Convert "/*!VersionNumber MySQL-specific-code */" to "MySQL-specific-code".
+// TODO: Find a better way:
+// 1. RegExpr is slow.
+// 2. Handle nested comment.
+func handleMySQLSpecificCode(sql string) string {
+	if strings.Index(sql, "/*!") == -1 {
+		// Fast way to check if text contains MySQL-specific code.
+		return sql
+	}
+	// Contains MySQL-specific code, convert it to normal sql text.
```
Comment (zimulala) [inline parser/yy_parser.go]: different subjects？

**Comment by shenli** [comment]
PTAL @zimulala

**Comment by zimulala** [comment]
LGTM

**Pull request merged**